internal/storage/types/global.go

```diff
@@ -0,0 +1,50 @@
+package types
+
+import (
+	"github.com/kubesimplify/ksctl/pkg/helpers/consts"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+)
+
+type StorageDocument struct {
+	ID primitive.ObjectID `bson:"_id,omitempty"`
+
+	// composite primary key ==> ClusterType, Region, ClusterName
+	ClusterType string `json:"cluster_type" bson:"cluster_type" `
+	Region      string `json:"region" bson:"region"`
+	ClusterName string `json:"cluster_name" bson:"cluster_name"`
+
+	InfraProvider     consts.KsctlCloud      `json:"cloud_provider" bson:"cloud_provider"`
+	BootstrapProvider consts.KsctlKubernetes `json:"bootstrap_provider"`
+
+	CloudInfra   *InfrastructureState      `json:"cloud_infrastructure_state" bson:"cloud_infrastructure_state,omitempty"`
+	K8sBootstrap *KubernetesBootstrapState `json:"kubernetes_bootstrap_state" bson:"kubernetes_bootstrap_state,omitempty"`
+
+	ClusterKubeConfig string `json:"cluster_kubeconfig" bson:"cluster_kubeconfig"` // TODO: should we use k8s.io/kops/pkg/kubeconfig or only string
+
+	//ClusterKubeConfig kubeconfig.KubectlConfig `json:"cluster_kubeconfig" bson:"cluster_kubeconfig"`
```

Comment:
@prateek041 should we use the kubernetes kubeconfig struct rather than string as type
becuase by this we can then open the scope of merging kubeconfig and removal of kubeconfig from users `~/.kube/config`

Reply:
Its is related to the Issue #5

Reply:
yep, we can use the type as *clusterconfig and can be used for various operations such as apps and the  autoscale client .